src/lib/animatedTransition/functions/createCompositeLoader.ts

```diff
@@ -0,0 +1,60 @@
+import { QueryClient } from "@tanstack/react-query";
+import { LoaderFunctionArgs } from "react-router-dom";
+import { setDelay } from "./delay";
+import { useTransitionStore } from "../stores/useTransitionStore";
+
+export type CompositeLoaderConfig = {
+  transitionType: "phased"; // @todo: add "continuous"
+  duration: number;
+};
+
+export type CompositeLoaderArgs = LoaderFunctionArgs &
+  CompositeLoaderConfig & {
+    url: URL;
+  };
+
+export type PageDataLoader<T extends object> = (
```

Comment:
```suggestion
export type PageDataLoader<T> = (
```

extends 없어도 되지 않을까요?!

1. `object` 는 좋은 타입이 아님
2. 굳이 T의 타입을 제한할 필요가 있을까?

Reply:
그러게요 굳이 오브젝트일 필요가 없을듯...?